CONTRIBUTING.md

```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
+If any of this sounds interesting, I hope you consider getting involved with Astro. Come say hi in the [**#new-contributors**][discord] channel on Discord anytime. We're always around and value contributions of any shape/size.
 
 # Contributor Manual
 
 ## Prerequisite
 
-```shell
-node: "^12.20.0 || ^14.13.1 || >=16.0.0"
-yarn: "^1.22.10"
-# otherwise, your build will fail
-```
+You’ll need the following tools set up on your system to contribute to this codebase:
 
-## Setting up your local repo
+- [Node 14.x or 16.x][install-node]
+- [Yarn 1.x][install-yarn] (_Note: we do NOT use Yarn 2!_)
+- [Git 2.x][install-git]
 
-Astro uses yarn workspaces, so you should **always run `yarn install` from the top-level project directory.** running `yarn install` in the top-level project root will install dependencies for `astro`, `www`, `docs`, and every package in the repo.
+## Setting up Astro locally
 
-```shell
-git clone && cd ...
-yarn install
-yarn build:all
+All work on Astro begins by downloading a copy locally. Start by [**forking**][how-to-fork] this repository (if you’re new to this, [GitHub has a great detailed guide][how-to-fork]). You should end up with an `astro/` code folder on your local machine.
+
+Switch to this folder by opening your **Terminal application** of choice and running:
+
+```
+cd /[path to my code folder]/astro
 ```
 
+Next, you’ll need to install dependencies and set up the project. You can do that by running the following commands from within the `astro/` folder:
+
+```
+yarn
+yarn build
+```
+
+This will download everything you need and build Astro locally for use. You’re ready to go!
+
 ## Development
 
-```shell
-# starts a file-watching, live-reloading dev script for active development
+While working within packages (such as `packages/astro/*`), you may notice you have to run `yarn build` to see your changes. This process can really slow you down if you are trying many changes at once. Try running the following instead:
+
+```
 yarn dev
-# build the entire project, one time.
+```
+
+This will start a build watcher, which will rebuild files as you work. Use this to speed up your workflow, and see your changes more rapidly.
+
+Of course, you can still build all files within a package, as before. You may need to do so from time to time:
+
+```
 yarn build
 ```
 
 ## Running tests
 
-```shell
-# run this in the top-level project root to run all tests
+You’ll need to run existing tests to make sure your work didn’t cause any regressions. To create a fresh build and run all tests, run the following from the project root:
+
+```
+yarn
+yarn build
 yarn test
-# run only a few tests, great for working on a single feature
-# (example - `yarn test rss` runs `astro-rss.test.js` tests)
-yarn test $STRING_MATCH
 ```
 
+If you find yourself wanting to not run the whole test suite, you can run the following instead:
+
+```
+cd packages/astro
+yarn test astro-markdown
```

Comment:
this actually works from the top level! you can remove `cd packages/astro`.

Reply:
also personal preference, but I feel like the existing snippet communicated the same info more clearly. Just a thought in case you're feeling the same!

Reply:
I’ll double-check; this used to not work from the top level

Reply:
Ah this was fixed! OK will revert the original.